src/activity_item_handling.cpp

```diff
@@ -536,14 +536,21 @@ static std::list<item> obtain_activity_items( player_activity &act, player &p )
         res.insert( res.begin(), excess.begin(), excess.end() );
     }
     // Load anything that remains (if any) into the activity
-    act.targets.clear();
-    act.values.clear();
-    if( !items.empty() ) {
+    if( !items.empty() && items.size() > act.targets.size() ) {
```

Comment:
```suggestion
    if( items.size() > act.targets.size() ) {
```
That check was redundant: if `items.size()` is 0 (meaning it's empty), it can't be greater than the size of another container.